src/code/z_demo.c

```diff
@@ -1232,684 +1406,776 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
-                        if (cmd->textId1 != 0xFFFF) {
-                            Message_ContinueTextbox(play, cmd->textId1);
+                        if (cmd->altTextId1 != CS_TEXT_ID_NONE) {
+                            Message_ContinueTextbox(play, cmd->altTextId1);
                         } else {
-                            csCtx->frames++;
+                            csCtx->curFrame++;
                         }
                     } else {
-                        if (cmd->textId2 != 0xFFFF) {
-                            Message_ContinueTextbox(play, cmd->textId2);
+                        if (cmd->altTextId2 != CS_TEXT_ID_NONE) {
+                            Message_ContinueTextbox(play, cmd->altTextId2);
                         } else {
-                            csCtx->frames++;
+                            csCtx->curFrame++;
                         }
                     }
                 }
 
                 if (dialogState == TEXT_STATE_9) {
-                    if (cmd->textId1 != 0xFFFF) {
-                        Message_ContinueTextbox(play, cmd->textId1);
+                    if (cmd->altTextId1 != CS_TEXT_ID_NONE) {
+                        Message_ContinueTextbox(play, cmd->altTextId1);
                     } else {
-                        csCtx->frames++;
+                        csCtx->curFrame++;
                     }
                 }
 
                 if (dialogState == TEXT_STATE_EVENT) {
                     if (Message_ShouldAdvance(play)) {
-                        func_8010BD58(play, cmd->base);
+                        func_8010BD58(play, cmd->textId);
                     }
                 }
             }
 
-            if (csCtx->frames == originalCsFrames) {
+            if (csCtx->curFrame == endFrame) {
                 Interface_ChangeHudVisibilityMode(HUD_VISIBILITY_NOTHING);
-                D_8011E1C0 = 0;
-                D_8011E1C4 = 0;
+                sCurTextId = 0;
+                sCurOcarinaAction = 0;
             }
         }
     }
 }
 
-void Cutscene_ProcessCommands(PlayState* play, CutsceneContext* csCtx, u8* cutscenePtr) {
+void Cutscene_ProcessScript(PlayState* play, CutsceneContext* csCtx, u8* script) {
     s16 i;
     s32 totalEntries;
     s32 cmdType;
     s32 cmdEntries;
-    CsCmdBase* cmd;
+    CsCmdGeneric* cmd;
     s32 cutsceneEndFrame;
     s16 j;
 
-    MemCpy(&totalEntries, cutscenePtr, 4);
-    cutscenePtr += 4;
-    MemCpy(&cutsceneEndFrame, cutscenePtr, 4);
-    cutscenePtr += 4;
+    MemCpy(&totalEntries, script, sizeof(totalEntries));
+    script += sizeof(totalEntries);
+
+    MemCpy(&cutsceneEndFrame, script, sizeof(cutsceneEndFrame));
+    script += sizeof(cutsceneEndFrame);
 
-    if ((cutsceneEndFrame < csCtx->frames) && (csCtx->state != CS_STATE_UNSKIPPABLE_EXEC)) {
-        csCtx->state = CS_STATE_UNSKIPPABLE_INIT;
+    if ((cutsceneEndFrame < csCtx->curFrame) && (csCtx->state != CS_STATE_RUN_UNSTOPPABLE)) {
```

Comment:
the corresponding argument in `CS_BEGIN_CUTSCENE` was renamed to "duration", so `cutsceneEndFrame` -> `csDuration` ?

also this matches (and I find it easier to read):
```suggestion
    if ((csCtx->curFrame > cutsceneEndFrame) && (csCtx->state != CS_STATE_RUN_UNSTOPPABLE)) {
```

Reply:
I think it makes more sense as "end frame" in the context of the checks it does against current frame, but its nice as duration from the perspective of the script itself (cause its just describing how long it is). its context dependent terms for the same thing

i did swap the condition tho

Reply:
I think using the same names in the macros and in the corresponding temps/struct members, is better for helping understanding the code and making the link

Alternatively `CS_BEGIN_CUTSCENE`'s `duration` could also be changed to `endFrame`, with a note on the macro that cutscenes start at frame 0

... btw where is the start frame set to 0?
it looks like it's from `csCtx->curFrame = 0xFFFF;` in `Cutscene_SetupScripted` then `csCtx->curFrame++;` in `CutsceneHandler_RunScript` bruh

Reply:
I understand your reasoning but personally still think its fine to have these two context dependent names
if there are more opinions that want it changed then I can do so

Reply:
What about `numFrames`, so it has better synergy with `startFrame`/`endFrame`, but is still similar to a `duration`? (or `frameCount`/`totalNumFrames`)

Reply:
I think id be ok with frameCount

Reply:
`frameCount` in the macro and `csFrameCount`/`cutsceneFrameCount` for the temp? Sure